### DISCUSSION TITLE
I'm getting DicomAssociationAbortedException, and I can't figure out why on DicomCFindRequest

### DISCUSSION THREAD

**Discussion by robertkjr3d** [fo-dicom/fo-dicom#1748] [Q&A]
When it hits the 'SendAsync()' it just dumps out of the procedure and shows the errors below in the output window.

```
                var worklistItems = new List<DicomDataset>();
                //PACS PATIENTID query
                var cf = DicomCFindRequest.CreatePatientQuery();
                cf.Dataset.AddOrUpdate(DicomTag.PatientID, szPatientIDsearch);
                cf.OnResponseReceived += (req, response) =>
                {
                    if (response.HasDataset)
                    {
                        worklistItems.Add(req.Dataset);
                    }
                };
                client = new Dicom.Network.Client.DicomClient(IPAddress, port, false, SendingAETitle, AETitle, 5000, 10000, 50, 5);
                await client.AddRequestAsync(cf);
                await client.SendAsync();

```
The output window is showing:
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in Dicom.Core.dll
Exception thrown: 'System.ObjectDisposedException' in System.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in Dicom.Core.dll
Exception thrown: 'Dicom.Network.DicomAssociationAbortedException' in mscorlib.dll
Exception thrown: 'System.ObjectDisposedException' in mscorlib.dll

If we are thinking it might be the server.  I'm able to put files on the server with another DICOM product.  It is a ORTHANC test-server.  Local within my home-network.

The FODicom binaries are 4.0.8 x86
The project is being built with the 4.8 framework.

**Comment by MrBean2016** [comment]
Are you query for worklist or for studies, your variable name suggest that? Either you add DicomUID.ModalityWorklistInformationModelFind or use  DicomCFindRequest.CreateWorklistQuery
Check out the samples in  the fo-dicom repo, everything you need is in there.

Orthanc also requires that you have enabled your calling AETitle in the config and added some worklist items if that is what you looking for.

**Reply by robertkjr3d** [reply]
My query did not seem to be incorrect.  I needed to change some settings on the test server.  So in that you were correct, that was causing the odd errors.  After I changed the settings on 'OTHANC', I was able to get the above code to retrieve results.

Although, when I ask for a similar cf.Dataset.AddOrUpdate(DicomTag.PatientName, szNameSearch) <-- I don't receive anything, and now I can't figure that one out.  And I've tried sending different parts of a known name that exists on the server.

**Reply by robertkjr3d** [reply]
I found that this command was necessary to find the names:
cf.Dataset.AddOrUpdate(DicomTag.PatientName, "{Last}^{First}");   <-- with that ^ character... Not sure why it was working like that.  I had not seen that in the examples.  I had another 'Client' had no trouble reading the server, so it seemed to be nothing to do with how the data was on the server.
